addons/suggest_locations/tests/test_stock_move_line.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+from . import common
+from ..models.suggest_locations_policy import SUGGEST_LOCATION_REGISTRY
+from odoo.exceptions import ValidationError
+
+
+class TestStockMoveLine(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockMoveLine, cls).setUpClass()
+        # Set policies
+        cls.picking_type_pick.u_suggest_locations_policy = "by_product"
+        cls.picking_type_goods_out.u_suggest_locations_policy = "match_move_line"
+        cls.ByProduct = SUGGEST_LOCATION_REGISTRY["by_product"](cls.env, False)
+        cls.ByMLS = SUGGEST_LOCATION_REGISTRY["match_move_line"](cls.env, False)
+        cls.MoveLine = cls.env["stock.move.line"]
+        # Create quants for picking and suggested location
+        cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+        cls.create_quant(cls.banana.id, cls.test_stock_location_02.id, 10)
+        cls.create_quant(cls.apple.id, cls.test_goodsout_location_01.id, 10)
+        cls.create_quant(cls.banana.id, cls.test_goodsout_location_02.id, 10)
+        cls.create_quant(cls.apple.id, cls.test_goodsout_location_03.id, 10)
+        # Create picking
+        cls._pick_info = [{"product": cls.banana, "qty": 5}, {"product": cls.apple, "qty": 5}]
+        cls.picking_id = cls.create_picking(
+            cls.picking_type_pick, products_info=cls._pick_info, confirm=True, assign=True
+        )
+        cls.mls = cls.picking_id.move_line_ids
+
+    def test01_check_correct_picking_type(self):
```

Comment:
Do we really need this test? We are setting this up. If this is a requirement for the other tests, you can replace this test for an aux function and call it at the start of the other tests.